DeviceSDK/src/main/java/org/astarteplatform/devicesdk/protocol/AstarteDeviceAggregateDatastreamInterface.java

```diff
@@ -28,4 +28,29 @@ public void streamData(String path, Map<String, Object> payload, DateTime timest
 
     transport.sendAggregate(this, path, payload, timestamp);
   }
+
+  public void validatePayload(String path, Map<String, Object> payload, DateTime timestamp)
+      throws AstarteInvalidValueException, AstarteInterfaceMappingNotFoundException {
+    String fomattedPath = path + "/";
+
+    for (Map.Entry<String, AstarteInterfaceMapping> interfaceMappingEntry :
+        getMappings().entrySet()) {
```

Comment:
Extract getMapping() function and store into local variable, reuse it also for the next for loop.